services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -0,0 +1,180 @@
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.SwitchChangeType
+import org.openkilda.model.SwitchId
+import org.openkilda.northbound.dto.v2.switches.PortHistoryResponse
+import org.openkilda.testing.service.northbound.NorthboundServiceV2
+
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+@Narrative("Verify that port history is created for the port up/down actions.")
+class PortHistorySpec extends HealthCheckSpecification {
+    @Autowired
+    NorthboundServiceV2 northboundV2
+
+    @Unroll
+    def "Port history are created for the port down/up actions when link is #islDescription"() {
+        given: "A link"
+        assumeTrue("Unable to find $islDescription ISL for this test", isl as boolean)
+        def timestampBefore = System.currentTimeSeconds()
+
+        when: "Execute port DOWN on the src switch"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "Port is really DOWN"
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LINK_DOWN"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        and: "Port history is created on the src switch"
+        Long timestampAfterDown = System.currentTimeSeconds()
+        with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+            it.size() == 1
+            checkHistoryPortDownAction(it[0], isl.srcSwitch.dpId, isl.srcPort)
+        }
+
+        when: "Execute port UP on the src switch"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "Port is really UP"
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LIVE"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+
+        and: "Port history is updated on the src switch"
+        Long timestampAfterUp = System.currentTimeSeconds()
+        with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterUp)) {
+            it.size() == 2
+            checkHistoryPortUpAction(it[0], isl.srcSwitch.dpId, isl.srcPort)
+        }
+
+        and: "Port history is not empty when link is direct"
+        northboundV2.getPortHistory(isl.dstSwitch.dpId, isl.dstPort, timestampBefore, timestampAfterUp).size() ==
+                historySizeOnDstSw
+
+        and: "Port history is also available using default timeline"
+        northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort).size() >= 2
```

Comment:
should be strict '2'? Why `>=`?

Reply:
Actually we are requesting port history information for default timeline(the day before timeTo)
some port history can be already generated for default timeline